independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Decorators.java

```diff
@@ -42,12 +42,13 @@ static DecoratorInfo createDecorator(ClassInfo decoratorClass, BeanDeployment be
 
         InjectionPointInfo delegateInjectionPoint = delegateInjectionPoints.get(0);
 
-        Integer priority = 0;
-        boolean priorityDeclared = false;
+        Integer priority = null;
         for (AnnotationInstance annotation : store.getAnnotations(decoratorClass)) {
             if (annotation.name().equals(DotNames.PRIORITY)) {
                 priority = annotation.value().asInt();
-                priorityDeclared = true;
+            }
+            if (priority == null && annotation.name().equals(DotNames.PRIORITY)) {
```

Comment:
You refer to `DotNames.PRIORITY` in the previous `if`, so this should be `DotNames.ARC_PRIORITY`?

Reply:
The change in `Interceptors` does have `ARC_PRIORITY`, which makes me confident that this should too.

Reply:
Good catch! TBH I don't expect anyone using `io.quarkus.arc.Priority` here but we should support this.